dvc/remote/gdrive.py

```diff
@@ -41,10 +41,43 @@ def __init__(self, path):
 def gdrive_retry(func):
     retry_re = re.compile(r"HttpError (403|500|502|503|504)")
 
+    def retry_error(exc):
+        import json
+
+        if RemoteGDrive.corpora != "drive" or not exc.args:
```

Comment:
are you sure we need this check? If we are getting rid of regexp then we need to get JSON, check HTTP code and make sure that it's not `file.permissions` + `forbidden`

Reply:
Since the response with permissions error we are trying to identify is not documented, I'm not sure if GDrive API has some others undocumented cases. `forbidden` reason and `403` error code seems can be generated in others cases.